### ISSUE TITLE
Incomplete deployment of changes between Umbraco Cloud environments when using the Umbraco Package

### ISSUE THREAD

**Issue by anh-duc-le** [skybrud/Skybrud.Umbraco.Redirects#69]
I have successfully installed the Umbraco Package v2.0.4 on my Umbraco Cloud Development environment, but I am seeing quite a few problems appear when I try to deploy these changes to my Staging environment.

## Failure to load redirects in Staging environment
In the redirect overview panel I am met with the following errors
- `Request error: The URL returned a 404 (not found): /umbraco/backoffice/SkyBrud/Redirects/GetRoodNodes`
- `Request error: The URL returned a 404 (not found): /umbraco/backoffice/SkyBrud/Redirects/GetRedirects`
- `Unable to load redirects: The list of redirects could not be loaded.`
- `Request error: The URL returned a 404 (not found): /umbraco/backoffice/SkyBrud/Redirects/GetRedirects`
- `Unable to load redirects: The list of redirects could not be loaded.`
![image](https://user-images.githubusercontent.com/37115195/89991759-9281df00-dc84-11ea-8caa-510619a11d12.png)

## Failure to add redirects in Staging environment
In the add panel I am faced with the following errors
- `Request error: The URL returned a 404 (not found): /umbraco/backoffice/SkyBrud/Redirects/AddRedirect
- `Saving failed: The redirect could not be saved due to an error on the server`
![image](https://user-images.githubusercontent.com/37115195/89992982-5cddf580-dc86-11ea-9230-158ad81c9b8d.png)

## Missing dbo.SkybrudRedirects table in the Staging database
The database table SkybrudRedirects only exists in the Development environment
![image](https://user-images.githubusercontent.com/37115195/89991442-230bef80-dc84-11ea-9dc3-6c01a3b79a92.png)

## Not appearing in the Installed packages page
The `\App_Data\packages\installedPackages.config` file on Staging was still empty after the deployment

```xml
<?xml version="1.0" encoding="utf-8"?>
<packages></packages>
```

And this causes the Umbraco Package to not show up in the backoffice
![image](https://user-images.githubusercontent.com/37115195/89990866-44b8a700-dc83-11ea-9cc0-659075d83838.png)

## Files that are not the issue
.dll files are not missing in `\bin`
![image](https://user-images.githubusercontent.com/37115195/89993889-a9760080-dc87-11ea-8d4a-da56328af22a.png)

Plugin files are not missing in `\App_Plugins\Skybrud.Umbraco.Redirects\`
![image](https://user-images.githubusercontent.com/37115195/89994017-d62a1800-dc87-11ea-9e7a-61a9126a7b08.png)

**Comment by abjerner** [comment]
Hi @anh-duc-le and sorry for the lack of reply,

You write that the DLLs are not missing. Is this in the stage environment or the local environment?

I'm not really familiar with Umbraco Cloud, but I'd say that the missing table and the 404 API calls are both indicators that `Skybrud.Umbraco.Redirects.dll` is missing in the stage environment. Could you re-check and confirm just to be sure?

**Comment by anh-duc-le** [comment]
Hi @abjerner  sorry for the late response. The DLLs in the screenshot are from the staging environment. So it seems that the files are properly pushed through the environments.

**Comment by abjerner** [comment]
I'm not really sure why it doesn't work then. The DLL *should* trigger a migration to run, which should then take of creating the table in the database.

Does the `umbracoKeyValue` table contain anything about the redirects package? If the migration has run, there should be an entry with `Umbraco.Core.Upgrader.State+Skybrud.Umbraco.Redirects` as the key. If it is there, you could try to delete that row, a reboot your website, and see what happens.

Also make sure to check the Umbraco log for anything that may hint what is going on 😉

**Comment by anh-duc-le** [comment]
I checked the `umbracoKeyValue` table and saw the following entry

![image](https://user-images.githubusercontent.com/37115195/95334490-7aa69000-08ae-11eb-91b0-e5b5b7f2d2d0.png)

I then looked for `dbo.SkybrudRedirects` and found it too!
![image](https://user-images.githubusercontent.com/37115195/95334484-77ab9f80-08ae-11eb-9d06-9fa619825615.png)

I went to install the package again on the Development environment and pushed the stages to the Staging environment.

After that I checked `\App_Data\packages\installedPackages.config` to find that it was now properly installed 

![image](https://user-images.githubusercontent.com/37115195/95335040-2cde5780-08af-11eb-8f1a-4b13b9414610.png)

In the backoffice it;s properly installed as well
![image](https://user-images.githubusercontent.com/37115195/95335202-64e59a80-08af-11eb-8593-a553548aafac.png)

I tried the CRUD actions on the redirects and they all worked fine too.

Seems like that it was a classic case of "turning it off and on again" by installing it once again.
